opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -177,6 +178,8 @@ def __setitem__(self, key, value):
                 elif (
                     self.maxlen is not None and len(self._dict) == self.maxlen
                 ):
+                    if not isinstance(self._dict, OrderedDict):
+                        self._dict = OrderedDict(self._dict)
```

Comment:
We only support Python 3.8 or greater, and since 3.7 [dictionaries keep the insertion order](https://mail.python.org/pipermail/python-dev/2017-December/151283.html). I don't think we need an `OrderedDict` at all.

Reply:
In next line, we do `self._dict.popitem(last=False)`.
`popitem(last)` is OrderedDict only method.
dict doesn't support efficient FIFO.

(BTW, I'm the author of Python 3.7 compact ordered dict. 😺 )

Reply:
Right, good point :v: